tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs

```diff
@@ -363,22 +378,32 @@ private async Task AddAllComputeSystemsFromAProvider(ComputeSystemsLoadedData da
     [RelayCommand]
     public void SearchHandler(string query)
     {
-        ComputeSystemCardsView.Filter = system =>
+        var currentProviders = PerProviderViewModels.ToList();
+        PerProviderViewModels.Clear();
+
+        var dontShowIndices = new List<int>();
+        for (var i = 0; i < currentProviders.Count; i++)
         {
-            if (system is CreateComputeSystemOperationViewModel createComputeSystemOperationViewModel)
+            var providerViewModel = currentProviders[i];
+            providerViewModel.SearchHandler(query);
+
+            if (!providerViewModel.IsVisible)
             {
-                return createComputeSystemOperationViewModel.EnvironmentName.Contains(query, StringComparison.OrdinalIgnoreCase);
+                dontShowIndices.Add(i);

```

Comment:
You don't really need this list of indexes anymore. Second loop can just iterate again and check for providerViewModel.IsVisible. This could be an optimization but probably not worth it for lists of small sizes.